src/schema.ts

```diff
@@ -40,36 +71,44 @@ export interface FieldSchema {
  *
  * @return a Schema object
  */
-export function build(data: any, opt: QueryConfig = {}): Schema {
+export function build(data: any, opt: QueryConfig = {}, tableSchema: TableSchema<TableSchemaFieldDescriptor> = {fields:[]}): Schema {
   opt = extend({}, DEFAULT_QUERY_CONFIG, opt);
 
   // create profiles for each variable
   let summaries: DLFieldProfile[] = summary(data);
   let types = inferAll(data); // inferAll does stronger type inference than summary
 
-  let fieldSchemas: FieldSchema[] = summaries.map(function(fieldProfile) {
-    let field: string = fieldProfile.field;
-    let primitiveType: PrimitiveType = types[field] as any;
+  let tableSchemaFieldIndex = tableSchema.fields.reduce((m, field: TableSchemaFieldDescriptor) => {
+    m[field.name] = field;
+    return m;
+  }, {});
+
+  let fieldSchemas: FieldSchema[] = summaries.map(function(fieldProfile, index) {
+
+    const name: string = fieldProfile.field;
+    const inIndex: number = index;
```

Comment:
Since inIndex is only used once below in `let fieldSchema` -- just refer to `index` directly there?